src/coredump/coredump.c

```diff
@@ -1342,14 +1349,64 @@ static int gather_pid_metadata_from_argv(
                         t = buf;
                 }
 
+                if (i == META_ARGV_PID) {
+                        /* Store this so that we can check whether the core will be forwarded to a container
+                         * even when the kernel doesn't provide a PIDFD. Can be dropped once baseline is
+                         * >= v6.16. */
+                        r = pidref_set_pidstr(&local_pidref, t);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to initialize pidref from pid %s: %m", t);
+                }
+
+                if (i == META_ARGV_PIDFD) {
+                        if (isempty(t))
```

Comment:
please add a comment here: `/* if the current kernel doesn't support the %F specifier (which resolves to a pidfd), but we included it in the core_pattern expression, we'll receive an empty string here. Deal with that gracefully. */`